clients/nodeRating/nodeRating.go

```diff
@@ -0,0 +1,164 @@
+package noderating
+
+import (
+	"encoding/json"
+	"fmt"
+	"math"
+
+	"github.com/ElrondNetwork/elrond-go-core/core/check"
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+	"github.com/ElrondNetwork/node-monitoring/clients"
+	"github.com/ElrondNetwork/node-monitoring/common"
+	"github.com/ElrondNetwork/node-monitoring/config"
+	"github.com/ElrondNetwork/node-monitoring/data"
+)
+
+var log = logger.GetOrCreate("clients/nodeRating")
+
+const (
+	firstRunValue    float64 = -1
+	maxRating                = 100
+	minReqTimeoutSec         = 1
+)
+
+const (
+	baseURL = "https://api.elrond.com"
+
+	nodesIdentifierPath = "/nodes?identity=%s"
+	nodesBLSKeyPath     = "/nodes/%s"
+)
+
+// ArgsNodeRating defines the arguments needed to create a new client
+type ArgsNodeRating struct {
+	Client clients.HTTPClient
+	Config *config.NodeRating
+}
+
+type nodeRating struct {
+	httpClient clients.HTTPClient
+	lastValues map[string]float64
+	firstRun   bool
+	config     *config.NodeRating
+}
+
+// NewNodeRatingClient creates an instance of httpClient which is a wrapper for http.Client
+func NewNodeRatingClient(args ArgsNodeRating) (*nodeRating, error) {
+	err := checkArgs(args)
+	if err != nil {
+		return nil, err
+	}
+
+	return &nodeRating{
+		httpClient: args.Client,
+		lastValues: make(map[string]float64),
+		firstRun:   true,
+		config:     args.Config,
+	}, nil
+}
+
+func checkArgs(args ArgsNodeRating) error {
+	if check.IfNil(args.Client) {
+		return ErrNilHTTPClient
+	}
+	if len(args.Config.PubKeys) == 0 {
+		return fmt.Errorf("no public keys provided in config")
+	}
+	if args.Config.Threshold <= 0 {
+		return fmt.Errorf("%w: invalid node rating threshold, provided %.2f", common.ErrInvalidValue, args.Config.Threshold)
+	}
+
+	return nil
+}
+
+func (hcw *nodeRating) GetEvent() (data.NotificationMessage, error) {
+	if hcw.firstRun == true {
+		log.Info("First run. Will not trigger any event.")
+		hcw.firstRun = false
+
+		return hcw.handleFirstRun()
```

Comment:
if this returns err the first time, shouldn't the firstRun be still true? 
could the firstRun-check be done on the len(hcw.lastValues) > 0?

Reply:
added default value for all `lastValues`, and also an additional check on the last value, this way we handle also nodes with default last values (in case an error appears when fetching the nodes)